### PR TITLE
feat: app settings integration

### PR DESCRIPTION
**Pull request by jhasuraj01**
## Todo

- [x] Integrate Analytics & Bug Report Toggle
- [x] Integrate cySync Update Toggle
- [x] Integrate reset cySync Dialog
- [x] Comment USB Diagnostics

## Screenshots:
Screenshot_20230918_124119

### CODE REVIEW

File: packages/cysync-core/src/pages/MainApp/Settings/Tabs/AppSettings.tsx [hunk 1]
```diff
@@ -20,6 +21,25 @@ export const AppSettings: React.FC = () => {
...
+
+  useEffect(() => {
+    keyValueStore.isAnalyticsAndBugReportEnabled.set(
+      isAnalyticsAndBugReportEnabled,
+    );
+  }, [isAnalyticsAndBugReportEnabled]);
+
+  useEffect(() => {
+    keyValueStore.isAutoUpdateCySyncEnabled.set(isAutoUpdateCySyncEnabled);
+  }, [isAutoUpdateCySyncEnabled]);
```
Comment (irshadCypherock) [inline packages/cysync-core/src/pages/MainApp/Settings/Tabs/AppSettings.tsx]: Keep the `useState` type as `boolean | undefined`. And if it is `undefined`, it means it hasn't been set yet by the  `.get()` function, in which case we don't set the variable on DB.

File: packages/cysync-core/src/dialogs/CySync/ResetCySync/context/index.tsx [hunk 2]
```diff
@@ -43,11 +47,19 @@ export const ResetCySyncDialogProvider: FC<ResetCySyncDialogProviderProps> = ({
...
     dispatch(closeDialog('resetCySync'));
   };
 
+  const onReset = async () => {
+    setLoading(true);
+    await getResetCySyncMethod()();
+    setLoading(false);
+    onClose();
+  };
+
```
Comment (irshadCypherock) [inline packages/cysync-core/src/dialogs/CySync/ResetCySync/context/index.tsx]: Reset CySync should require a password input, if password is set.

File: packages/cysync-core/src/dialogs/CySync/ResetCySync/context/index.tsx [hunk 3]
```diff
@@ -43,11 +47,19 @@ export const ResetCySyncDialogProvider: FC<ResetCySyncDialogProviderProps> = ({
   const lang = useAppSelector(selectLanguage);
   const dispatch = useAppDispatch();
   const deviceRequiredDialogsMap: Record<number, number[] | undefined> = {};
+  const [loading, setLoading] = useState<boolean>(false);
```
Comment (irshadCypherock) [inline packages/cysync-core/src/dialogs/CySync/ResetCySync/context/index.tsx]:
```suggestion
  const [isLoading, setIsLoading] = useState<boolean>(false);
```

**Review by irshadCypherock — CHANGES_REQUESTED** [review]

File: packages/cysync-core/src/dialogs/CySync/ResetCySync/context/index.tsx (hunk 2 above)
Comment (jhasuraj01) [reply packages/cysync-core/src/dialogs/CySync/ResetCySync/context/index.tsx]: Password Input requirement is changed and is no longer required. Verified by design team.

File: packages/cysync-core/src/dialogs/CySync/ResetCySync/context/index.tsx (hunk 3 above)
Comment (jhasuraj01) [reply packages/cysync-core/src/dialogs/CySync/ResetCySync/context/index.tsx]: updated variable name

File: packages/cysync-core/src/pages/MainApp/Settings/Tabs/AppSettings.tsx (hunk 1 above)
Comment (jhasuraj01) [reply packages/cysync-core/src/pages/MainApp/Settings/Tabs/AppSettings.tsx]: Implemented and added loading state to `Toggle` Component

**Review by irshadCypherock — APPROVED** [review]

**Pull request merged**